web3-android-mpc-provider/src/main/java/com/web3auth/web3_android_mpc_provider/EthereumTssAccount.java

```diff
@@ -59,11 +61,20 @@ public String signMessage(String message) throws TSSClientError, CustomSigningEr
         return TSSHelpers.hexSignature(signatureResult.getFirst(), signatureResult.getSecond(), v);
     }
 
-    //todo: Method for signing eip712 structured data
+    public String signTypedData(String jsonData) throws IOException, TSSClientError, CustomSigningError {
+        StructuredDataEncoder dataEncoder = new StructuredDataEncoder(jsonData);
+        byte[] hashStructuredData = dataEncoder.hashStructuredData();
+        String structuredData = android.util.Base64.encodeToString(hashStructuredData, Base64.NO_WRAP);
+        Triple<BigInteger, BigInteger, Byte> signatureResult = sign(structuredData);
+        Byte v = signatureResult.getThird();
+        if (v < 27) {
```

Comment:
There's another comment about this, check should be able to be removed, since we know v is in (0,1). 

Here it may be better to keep it for clarity when reading code.
```suggestion
```